dash/frontend/src/app/modules/shared/alert-dialog/alert-dialog.component.ts

```diff
@@ -30,15 +30,16 @@ export class AlertDialogComponent {
             if (!environment.production) {
               console.log(result);
             }
+
```

Comment:
If there's no function to run, will this still close?
atm it looks like if there's no function, it won't close at all.

Reply:
You may be right, I will verify tomorrow

Reply:
The modal always closed, because the submit button had the mat-dialog-close property. In the latest commit, I made it so it will run this section either after the observable to run, or if there is no observeable.

I checked all the spots that it is used and it works as I would expect in each.

I think that this alert dialog should probably be made to only have 1 "mode" as a tech debt item, as right now, it can be running an observeable or not, and can navigate to a page, refresh the page, or do neither.

It probably should just return the boolean and let the calling page handle making the observable calls and any navigation that needs to happen as a result afterwards.